client/taskenv/env_test.go

```diff
@@ -650,17 +650,16 @@ func TestEnvironment_AppendHostEnvvars(t *testing.T) {
 	}
 }
 
-// TestEnvironment_DashesInTaskName asserts dashes in port labels are properly
-// converted to underscores in environment variables.
-// See: https://github.com/hashicorp/nomad/issues/2405
```

Comment:
So I think we need to continue to support this behavior (creating environment variables based on the job id with underscores) at the same time as we relax the restriction that renames hyphenated variables. What's your usecase here? do you want to permit variables explicitly define with hyphenated names, or change the generated `NOMAD_ADDR_job_name` variables?

Reply:
I want to support environment variables defined with hyphenated names, I'll change the PR to keep the cleanup of generated variables